activerecord/test/cases/associations_test.rb

```diff
@@ -502,6 +505,22 @@ def test_preload_grouped_queries_with_already_loaded_records
     end
   end
 
+  def test_preload_grouped_queries_with_already_loaded_through_records
+    member = members(:groucho)
+    expected_member_detail_ids = member.organization_member_details_2.pluck(:id)
+
+    member.reload
+    member.organization # load through record
+
+    assert_queries(1) do
+      ::ActiveRecord::Associations::Preloader.new(records: [member], associations: :organization_member_details_2).call
+    end
+
+    assert_no_queries do
+      assert_equal expected_member_detail_ids.sort, member.organization_member_details_2.map(&:id).sort
```

Comment:
This fails without the fix above
<img width="1158" alt="Screenshot 2022-11-30 at 6 39 00 pm" src="https://user-images.githubusercontent.com/54629302/204804629-e32f23a2-eaf7-4b6b-ad90-0a78eac49867.png">